certgen/certgen_test.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certgen
+
+import (
+	"bytes"
+	"crypto"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/monologue/interval"
+	"github.com/google/trillian/crypto/keys/pem"
+)
+
+const (
+	rootFile    = "./testdata/test_ca.pem"
+	rootKeyFile = "./testdata/test_ca.key"
+
+	subjectCommonName         = "test-leaf-certificate"
+	subjectOrganization       = "Test Organisation"
+	subjectOrganizationalUnit = "Test Organisational Unit"
+	subjectLocality           = "Test Locality"
+	subjectCountry            = "GB"
+	signatureAlgorithm        = x509.SHA256WithRSA
+	prefix                    = "test-log"
+)
+
+func rootAndKeySetup(rootFile, rootKeyFile string) (*x509.Certificate, crypto.Signer, error) {
+	rootPEM, err := ioutil.ReadFile(rootFile)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading root cert: %s", err)
+	}
+	root, err := x509util.CertificateFromPEM(rootPEM)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error parsing root cert: %s", err)
+	}
+
+	rootKeyPEM, err := ioutil.ReadFile(rootKeyFile)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading root key: %s", err)
+	}
+	rootKey, err := pem.UnmarshalPrivateKey(string(rootKeyPEM), "")
+	if err != nil {
+		return nil, nil, fmt.Errorf("error parsing root key: %s", err)
+	}
+
+	return root, rootKey, nil
+}
+
+func TestIssueCertificate(t *testing.T) {
+	tests := []struct {
+		desc             string
+		notAfterInterval *interval.Interval
+	}{
```

Comment:
it's a bit odd to have no `want` fields here (/muses phad)

Reply:
yeah ... I gave some thought to whether you'd want to write 'wantConfig' or similar, but it would be a bit contorted and not worth it just to follow convention.